<launch>

  <include file="$(find ros_tcp_endpoint)/launch/endpoint.launch"/>
  <include file="$(find yolov7_ros)/launch/yolov7_multi.launch"/>

  <!-- Outside the field -->
  <!-- <node pkg="rostopic" type="rostopic" name="pose_publisher1" args="pub -1 /agent_1/cmd/pose geometry_msgs/Pose '{position: {x: -3.354, y: -2.0, z: 2.350}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}'" />
  <node pkg="rostopic" type="rostopic" name="pose_publisher2" args="pub -1 /agent_2/cmd/pose geometry_msgs/Pose '{position: {x: -3.354, y: -7.3, z: 2.350}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}'" />
  <node pkg="rostopic" type="rostopic" name="pose_publisher3" args="pub -1 /agent_3/cmd/pose geometry_msgs/Pose '{position: {x: -3.354, y: -14.1, z: 2.350}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}'" /> -->

  <!-- Inside the field -->
  <node pkg="rostopic" type="rostopic" name="pose_publisher1" args="pub -1 /agent_1/cmd/pose geometry_msgs/Pose '{position: {x: 1.8, y: -3.2, z: 2.350}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}'" />
  <node pkg="rostopic" type="rostopic" name="pose_publisher2" args="pub -1 /agent_2/cmd/pose geometry_msgs/Pose '{position: {x: 14.04, y: -1.35, z: 2.350}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}'" />
  <node pkg="rostopic" type="rostopic" name="pose_publisher3" args="pub -1 /agent_3/cmd/pose geometry_msgs/Pose '{position: {x: 2.07, y: -13.4, z: 2.350}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}'" />

  <!-- Clustering -->
  <node name="Assignment" pkg="nmpc_ros" type="kmeans.py" output="screen">
        <param name="n_agents" value="3" />
  </node> 

</launch>